src/server/scripts/Battlegrounds/TempleOfKotmogu/battleground_temple_of_kotmogu.cpp

```diff
@@ -0,0 +1,630 @@
+        UpdateWorldState(orbData->WorldStateAlliance, 0);
+        UpdateWorldState(orbData->WorldStateHorde, 0);
+        SpawnOrb(orbData);
+        battlegroundMap->UpdateSpawnGroupConditions();
+
+        if (Creature const* trigger = battlegroundMap->GetCreature(_triggerGuid))
+            trigger->AI()->Talk(orbData->TriggerText, player);
+    }
+
+    void DoAction(uint32 actionId, WorldObject* source, WorldObject* target) override
+    {
+        switch (actionId)
+        {
+            case TempleOfKotmogu::Actions::CheckScore:
+                // ReSharper disable once CppExpressionWithoutSideEffects
+                CheckWinner();
+                break;
+            case TempleOfKotmogu::Actions::ConsumeAreaTrigger:
+                HandleConsumeAreaTrigger(Object::ToAreaTrigger(source), Object::ToPlayer(target));
+                break;
+            default:
+                break;
+        }
+    }
+
+    void HandleConsumeAreaTrigger(AreaTrigger* trigger, Player* player)
+    {
+        // find index
+        std::size_t index = 0;
+        for (std::size_t i = 0; i < _healingBuffs.size(); i++)
+        {
+            if (_healingBuffs[i] == trigger->GetGUID())
+            {
+                index = i;
+                break;
+            }
+        }
+
+        player->CastSpell(player, TempleOfKotmogu::Spells::Restoration, true);
+        trigger->Remove();
+
+        _scheduler.Schedule(90s, [&, index](TaskContext)
+        {
+            SpawnHealingBuff(index);
+        });
+    }
+
+    bool CheckWinner() const
+    {
+        uint32 const maxScore = sWorldStateMgr->GetValue(TempleOfKotmogu::WorldStates::MaxPoints, battlegroundMap);
+        if (battleground->GetTeamScore(TEAM_HORDE) == maxScore && battleground->GetTeamScore(TEAM_ALLIANCE) == maxScore)
+        {
+            UpdateWorldState(TempleOfKotmogu::WorldStates::AlliancePoints, maxScore);
+            UpdateWorldState(TempleOfKotmogu::WorldStates::HordePoints, maxScore);
+            battleground->EndBattleground(TEAM_OTHER);
+            return true;
+        }
+
+        if (battleground->GetTeamScore(TEAM_HORDE) == maxScore)
+        {
+            UpdateWorldState(TempleOfKotmogu::WorldStates::HordePoints, maxScore);
+            battleground->EndBattleground(HORDE);
+            return true;
+        }
+
+        if (battleground->GetTeamScore(TEAM_ALLIANCE) == maxScore)
+        {
+            UpdateWorldState(TempleOfKotmogu::WorldStates::AlliancePoints, maxScore);
+            battleground->EndBattleground(ALLIANCE);
+            return true;
+        }
+
+        return false;
+    }
+
+    void SpawnHealingBuff(std::size_t index)
+    {
+        if (Creature* creature = battlegroundMap->GetCreature(_triggerGuid))
+            if (AreaTrigger const* trigger = AreaTrigger::CreateAreaTrigger({ TempleOfKotmogu::AreaTriggerCreateProperties::PvpRuneRejuv, true }, TempleOfKotmogu::Positions::HealingBuffs[index], -1, creature, nullptr))
```

Comment:
There's no indication who's the caster of this areatrigger in any sniff?

Reply:
I'm blind, I saw the creature pointer being passed now at the end of the line >:

Reply:
The actual summoner is an `AIGroup`. I just used the creature as convenience